src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -539,35 +539,20 @@ private MethodSymbol GetArgumentInfoFactory()
         // however
         //     dynamic d = ...;
         //     GetS().M(d); // becomes Site(GetS(), d) without ref on the target obj arg
-        internal static RefKind GetReceiverRefKind(BoundExpression loweredReceiver)
+        internal RefKind GetReceiverRefKind(BoundExpression loweredReceiver)
         {
             Debug.Assert(loweredReceiver.Type is { });
             if (!loweredReceiver.Type.IsValueType)
             {
                 return RefKind.None;
             }
 
-            switch (loweredReceiver.Kind)
-            {
-                case BoundKind.Parameter:
-                    Debug.Assert(!LocalRewriter.IsCapturedPrimaryConstructorParameter(loweredReceiver));
-                    goto case BoundKind.Local;
-
-                case BoundKind.Local:
-                case BoundKind.ArrayAccess:
-                case BoundKind.ThisReference:
-                case BoundKind.PointerIndirectionOperator:
-                case BoundKind.PointerElementAccess:
-                case BoundKind.RefValueOperator:
-                    return RefKind.Ref;
-
-                case BoundKind.BaseReference:
-                // base dynamic dispatch is not supported, an error has already been reported
-                case BoundKind.TypeExpression:
-                    throw ExceptionUtilities.UnexpectedValue(loweredReceiver.Kind);
-            }
-
-            return RefKind.None;
+            var hasHome = Binder.HasHome(loweredReceiver,
+                Binder.AddressKind.Writeable,
+                _factory.CurrentFunction,
+                _factory.Compilation.IsPeVerifyCompatEnabled,
```

Comment:
>_factory.Compilation.IsPeVerifyCompatEnabled [](http://example.com/codeflow?start=16&length=44)

I think we don't need to depend on this setting here. The most permissive value is probably be the closest match with the previous behavior.